corgi/core/models.py

```diff
@@ -1774,25 +1775,30 @@ def get_roots(self, using: str = "read_only") -> QuerySet[ComponentNode]:
 
     @property
     def cpes(self) -> QuerySet:
-        """Build and return a list of CPEs from all Variants this Component relates to"""
+        """Build and return a list of CPEs this Component relates to"""
         # For each Variant-type relation, get the linked Variant's CPE directly
         # Remove any duplicates, and return the CPEs in sorted order so manifests are stable
-        return (
-            self.productvariants.exclude(cpe="")
-            .values_list("cpe", flat=True)
-            .distinct()
-            .order_by("cpe")
-        )
-
-        # For Stream-type relations, we no longer link any Variants
-        # since this caused data quality issues, where Components were linked to Variants
-        # that didn't actually ship those Components
-        # We could look at all the linked streams to get CPEs from all their child Variants
-        # But this would cause the same problem - we'd report too many Variants / incorrect CPEs
+        if self.productvariants.exists():
+            return (
+                self.productvariants.exclude(cpe="")
+                .values_list("cpe", flat=True)
+                .distinct()
+                .order_by("cpe")
+            )
+        elif self.software_build:
+            build_variants = (
+                ProductComponentRelation.objects.filter(software_build=self.software_build)
+                .values_list("product_ref", flat=True)
+                .distinct()
+            )
+            if build_variants:
+                return (
+                    CollectorErrataProductVariant.objects.filter(name__in=build_variants)
+                    .values_list("cpe", flat=True)
+                    .order_by("cpe")
```

Comment:
ProductVariant models don't exist for brew_tag streams (like openshift-4.*) which we are targeting here. We would have make a bunch of other changes to the code to allow that linking. For example, setting up a many-to-many relationship between stream and variant to allow variants to be linked to multiple streams. Also changes to the save_product_taxonomy functions to prevent builds linked to ProductVariants matched via brew_tags from being also linked to ProductStreams. 

Using the Collector models here seems like a good compromise to me. Although if we really wanted to be strict about that we should probably allow ProductVariant models to have their productstreams, etc be null so that we can create ProductVariant and use those parentless ProductVariants in situations like this.